api/src/wfl/module/covid.clj

```diff
@@ -303,6 +229,139 @@
+    (catch Throwable t
+      (throw (ex-info "Cannot access the workspace" {:workspace workspace
+                                                     :cause     (.getMessage t)})))))
+
+(defn ^:private add-continuous-workload-record
+  "Use `tx` and workload `id` to create a \"ContinuousWorkload\" instance and
+  return the ID of the ContinuousWorkload."
+  [tx id {:keys [source sink executor] :as _request}]
+  (let [set-details "UPDATE
+                         ContinuousWorkload
+                     SET
+                         source_type   = ?::source,
+                         executor_type = ?::executor,
+                         sink_type     = ?::sink
+                     WHERE
+                         id = ? "
+        src-exc-snk [(create-source! tx id source)
+                     (create-executor! tx id executor)
+                     (create-sink! tx id sink)]
+        items       (->> (map second src-exc-snk)
+                         (zipmap [:source_items :executor_items :sink_items])
+                         (jdbc/insert! tx :ContinuousWorkload)
+                         first
+                         :id)]
+    (jdbc/execute! tx (concat [set-details] (map first src-exc-snk) [items]))
+    items))
+
+(defn create-covid-workload [tx request]
+  (verify-source! (get-in request [:source]))
+  (verify-executor! (get-in request [:executor]))
+  (verify-sink! (get-in request [:sink]))
+  (let [set-pipeline "UPDATE workload
+                      SET pipeline = ?::pipeline
+                      WHERE id = ?"
+        id           (add-workload-record tx request)
+        items        (add-continuous-workload-record tx id request)]
+    (jdbc/execute! tx [set-pipeline pipeline id])
+    (jdbc/update! tx :workload {:items items} ["id = ?" id])
+    (workloads/load-workload-for-id tx id)))
+
+(defn update-covid-workload
+  "Use transaction TX to update WORKLOAD statuses."
+  [tx {:keys [started finished] :as workload}]
+  (letfn [(update-workload-status [])
+          (update! [{:keys [id source executor sink] :as _workload}]
+            (-> (update-source! source)
+                (update-executor! executor)
+                (update-sink! sink))
+            (update-workload-status)
+            (workloads/load-workload-for-id tx id))]
+    (if (and started (not finished)) (update! workload) workload)))
+
+(defn load-covid-workload-impl [tx {:keys [items] :as workload}]
+  (if-let [id (util/parse-int items)]
+    (let [details (load-record-by-id! tx "ContinuousWorkload" id)]
+      (->> {:source   (load-source! tx details)
+            :executor (load-executor! tx details)
+            :sink     (load-sink! tx details)}
+           (merge workload)
+           (filter second)
+           (into {})))
+    (throw (ex-info "Invalid ContinuousWorkload identifier"
+                    {:id       items
+                     :workload workload}))))
+
 (defoverload create-sink! terra-workspace-sink-name create-terra-workspace-sink)
```

Comment:
These `create-sink!` and `load-sink!` dispatches should be bumped up to live with the sink method definitions, below `load-terra-workspace-sink`.

Reply:
Done :)